manifests/init.pp

```diff
@@ -45,38 +45,52 @@
 class rhsm (
   $rh_user,
   $rh_password,
-  $servername = 'subscription.rhn.redhat.com',
+  $servername     = 'subscription.rhn.redhat.com',
+  $pool           = undef,
   $proxy_hostname = undef,
-  $proxy_port = undef,
-  $proxy_user = undef,
+  $proxy_port     = undef,
+  $proxy_user     = undef,
   $proxy_password = undef,
-  $baseurl= 'https://cdn.redhat.com',
-  $manage_repos = 1,
+  $baseurl        = 'https://cdn.redhat.com',
+  $package_ensure = 'latest'
 ) {
 
   if $proxy_hostname {
-    $proxycli = "--proxy=http://${proxy_hostname}:${proxy_port} --proxyuser=${proxy_user} --proxypass=${proxy_password}"
+    if $proxy_user and $proxy_password {
+      $proxycli = "--proxy=http://${proxy_hostname}:${proxy_port} --proxyuser=${proxy_user} --proxypass=${proxy_password}"
+    } else {
+      $proxycli = "--proxy=http://${proxy_hostname}:${proxy_port}"
+    }
   } else {
     $proxycli = ''
   }
 
+  if $pool == undef {
+    $command = "/usr/sbin/subscription-manager register --name=\"${::fqdn}\"  --username=\"${rh_user}\" --password=\"${rh_password}\" --auto-attach ${proxycli} && /usr/sbin/subscription-manager repo-override --repo rhel-${::operatingsystemmajrelease}-server-optional-rpms --add=enabled:1 && /usr/sbin/subscription-manager repo-override --repo rhel-${::operatingsystemmajrelease}-server-extras-rpms --add=enabled:1"
+  } else {
+    $command = "/usr/sbin/subscription-manager register --name=\"${::fqdn}\"  --username=\"${rh_user}\" --password=\"${rh_password}\" ${proxycli} && /usr/sbin/subscription-manager attach --pool=${pool}"
```

Comment:
The --force param is unpredictable and has the potential to cause more problems than solving them.